lms/djangoapps/branding/api.py

```diff
@@ -170,50 +170,107 @@ def _footer_social_links():
+
+    if language == settings.LANGUAGE_CODE:
+        links.append(("media_kit", (marketing_link("MEDIA_KIT"), _("Media Kit"))))
+        links.append(("donate", (marketing_link("DONATE"), _("Donate"))))
 
     return [
         {
             "name": link_name,
             "title": link_title,
             "url": link_url,
         }
-        for link_name, link_url, link_title in [
-            ("blog", marketing_link("BLOG"), _("Blog")),
-            ("contact", _build_support_form_url(), _("Contact Us")),
-            ("help-center", settings.SUPPORT_SITE_LINK, _("Help Center")),
-            ("media_kit", marketing_link("MEDIA_KIT"), _("Media Kit")),
-            ("donate", marketing_link("DONATE"), _("Donate")),
-        ]
+        for link_name, link_url, link_title in links
         if link_url and link_url != "#"
     ]
 
 
-def _build_support_form_url():
-    return '{base_url}/support/contact_us'.format(base_url=settings.LMS_ROOT_URL)
+def _find_position_of_link(links, key):
+    "Returns position of the link to be inserted"
+    for link in links:
+        if link[0] == key:
+            return links.index(link) + 1
 
 
-def _footer_navigation_links():
+def _footer_navigation_links(language=settings.LANGUAGE_CODE):
     """Return the navigation links to display in the footer. """
     platform_name = configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME)
+    links = [
+        ("about", (marketing_link("ABOUT"), _("About"))),
+        ("enterprise", (marketing_link("ENTERPRISE"),
+                        _("{platform_name} for Business").format(platform_name=platform_name))),
+        ("blog", (marketing_link("BLOG"), _("Blog"))),
+        ("help-center", (settings.SUPPORT_SITE_LINK, _("Help Center"))),
+        ("contact", (_build_support_form_url(), _("Contact"))),
+        ("careers", (marketing_link("CAREERS"), _("Careers"))),
+        ("donate", (marketing_link("DONATE"), _("Donate"))),
+    ]
+
+    if language == settings.LANGUAGE_CODE:
+        position = _find_position_of_link(links, 'blog')
+        links.insert(position, ("news", (marketing_link("NEWS"), _("News"))))
+
     return [
         {
             "name": link_name,
             "title": link_title,
             "url": link_url,
         }
-        for link_name, link_url, link_title in [
-            ("about", marketing_link("ABOUT"), _("About")),
-            ("enterprise", marketing_link("ENTERPRISE"),
-             _("{platform_name} for Business").format(platform_name=platform_name)),
-            ("blog", marketing_link("BLOG"), _("Blog")),
-            ("news", marketing_link("NEWS"), _("News")),
-            ("help-center", settings.SUPPORT_SITE_LINK, _("Help Center")),
-            ("contact", reverse("support:contact_us"), _("Contact")),
-            ("careers", marketing_link("CAREERS"), _("Careers")),
-            ("donate", marketing_link("DONATE"), _("Donate")),
-        ]
+        for link_name, link_url, link_title in links
```

Comment:
Same thing here, the unpack fails with the given input.

```suggestion
        for link_name, (link_url, link_title) in links
```